src/pages/index.js

```diff
@@ -17,131 +21,185 @@ import ModalWithForm from "../components/ModalWithForm.js";
+        deleteForm.open();
+      },
+      handleCardLike: (id, isLiked) => {
+        if (isLiked) {
+          api
+            .removeLike(id)
+            .then((res) => {
+              card.updateLikes(res.likes);
+            })
+            .catch((err) =>
+              console.log(`An error occurred when removing a like: ${err}`)
+            );
+        } else {
+          api
+            .addLike(id)
+            .then((res) => {
+              card.updateLikes(res.likes);
+            })
+            .catch((err) =>
+              console.log(`An error occurred when adding a like: ${err}`)
+            );
+        }
+      },
+      cardSelector: selectors.cardTemplate
+    },
+    
+  );
+  return card.getView();
+};
 
 // Instance of the UserInfo class
-const userInfo = new UserInfo({
-  userNameSelector: ".profile__name",
-  userDescriptionSelector: ".profile__description",
-  userImageSelector: ".profile__image",
+const newUserInfo = new UserInfo({
+  nameSelector: "#profileName",
+  descSelector: "#profileDescription",
+  avatarSelector: "#profileImage",
 });
 
-// Create and enable form validators for editing profile and adding cards
-const editProfileValidator = new FormValidator(config, profileForm);
-editProfileValidator.enableValidation();
-const editCardValidator = new FormValidator(config, cardAddModal);
-editCardValidator.enableValidation();
+// get web server info, cards and user data.
+api
+  .getWebpageInfo()
+  .then(([cardData, userData]) => {
+    userId = userData._id;
+    cardSection = new Section(
+      {
+        items: cardData,
+        renderer: (data) => {
+          const cardElement = card(data);
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".cards"
+    );
+    cardSection.renderItems();
+    newUserInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+    });
+    newUserInfo.setAvatar(userData.avatar);
+  })
+  .catch((err) =>
+    console.log(
+      `An error occurred when loading initial user and card data: ${err}`
+    )
+  );
+
+const cardAddForm = new ModalWithForm("#cardAddModal", (data) => {
+  const newCard = { name: data['card-title-input'], link: data['card-image-input'] };
+  cardAddForm.setApiCalling(true);
+  api
+    .addCard(newCard)
+    .then((result) => {
+      const cardElement = card(result);
+      cardSection.addItem(cardElement);
+      cardAddForm.close();
+    })
+    .catch((err) =>
+      console.log(`An error occurred when loading new card data: ${err}`)
+    )
+    .finally(() => cardAddForm.setApiCalling(false));
+});
+cardAddForm.setEventListeners();
 
-// Create an instance of the Section class
-const section = new Section(
-  { items: initialCards, renderer: (item) => renderCard(item, section) },
-  ".cards"
+addCardButton.addEventListener("click", () => {
+  addFormValidator._toggleButtonState();
+  cardAddForm.open();
+});
+const addFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```